crates/ads126x/src/lib.rs

```diff
@@ -0,0 +1,188 @@
+#![no_std]
+#![no_main]
+
+mod error;
+mod register;
+
+use error::ADS126xError;
+use register::{Register, IdRegister, PowerRegister, InterfaceRegister};
+
+use bitflags::bitflags;
+use embedded_hal::spi::FullDuplex;
+use heapless::Vec;
+
+pub struct ADS126x<SPI>
+where
+    SPI: FullDuplex<u8>,
+{
+    spi: SPI,
+}
+
+bitflags! {
+    pub struct StatusRegister: u8 {
+        const ADC2      = 0b1000_0000;
+        const ADC1      = 0b0100_0000;
+        const EXTCLK    = 0b0010_0000;
+        const REF_ALM   = 0b0001_0000;
+        const PGAL_ALM  = 0b0000_1000;
+        const PGAH_ALM  = 0b0000_0100;
+        const PGAD_ALM  = 0b0000_0010;
+        const RESET     = 0b0000_0001;
+    }
+}
+
+impl StatusRegister {
+    pub fn is_adc2_data_new(&self) -> bool {
+        self.contains(StatusRegister::ADC2)
+    }
+
+    pub fn is_adc1_data_new(&self) -> bool {
+        self.contains(StatusRegister::ADC1)
+    }
+
+    pub fn is_extclk(&self) -> bool {
+        self.contains(StatusRegister::EXTCLK)
+    }
+
+    pub fn is_ref_alm(&self) -> bool {
+        self.contains(StatusRegister::REF_ALM)
+    }
+
+    pub fn is_pgal_alm(&self) -> bool {
+        self.contains(StatusRegister::PGAL_ALM)
+    }
+
+    pub fn is_pgah_alm(&self) -> bool {
+        self.contains(StatusRegister::PGAH_ALM)
+    }
+
+    pub fn is_pgad_alm(&self) -> bool {
+        self.contains(StatusRegister::PGAD_ALM)
+    }
+
+    pub fn is_reset(&self) -> bool {
+        self.contains(StatusRegister::RESET)
+    }
+}
+
+pub enum ADCCommand {
+    NOP,
+    RESET,
+    START1,
+    STOP1,
+    START2,
+    STOP2,
+    RDATA1,
+    RDATA2,
+    SYOCAL1,
+    SYGCAL1,
+    SFOCAL1,
+    SYOCAL2,
+    SYGCAL2,
+    SFOCAL2,
+    RREG(Register, u8), // (register address, number of registers)
+    WREG(Register, u8), // (register address, number of registers)
+}
+
+impl<SPI> ADS126x<SPI>
+where
+    SPI: FullDuplex<u8>,
+{
+    pub fn new(spi: SPI) -> Self {
+        Self { spi }
+    }
+
+    pub fn send_command(&mut self, command: ADCCommand) -> Result<(), ADS126xError> {
+        let (opcode1, opcode2) = match command {
+            ADCCommand::NOP => (0x00, None),
+            ADCCommand::RESET => (0x06, None),
+            ADCCommand::START1 => (0x08, None),
+            ADCCommand::STOP1 => (0x0A, None),
+            ADCCommand::START2 => (0x0C, None),
+            ADCCommand::STOP2 => (0x0E, None),
+            ADCCommand::RDATA1 => (0x12, None),
+            ADCCommand::RDATA2 => (0x14, None),
+            ADCCommand::SYOCAL1 => (0x16, None),
+            ADCCommand::SYGCAL1 => (0x17, None),
+            ADCCommand::SFOCAL1 => (0x19, None),
+            ADCCommand::SYOCAL2 => (0x1B, None),
+            ADCCommand::SYGCAL2 => (0x1C, None),
+            ADCCommand::SFOCAL2 => (0x1E, None),
+            ADCCommand::RREG(addr, num) => (0x20 | addr as u8, Some(num)),
```

Comment:
Is it possible to do addr.into() instead of a cast? No real difference in the output but I'm generally not a fan of casting like this (no real reason, more semantics).

Reply:
I could, but we'd need to `impl Into<u8> for Register` which will essentially just cast it to a u8 anyway (since we have `#[repr(u8)]` and it's guaranteed to be a u8. Do you want me to do so?

Reply:
I think the cast is fine then since you've restricted it already.